src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -181,33 +199,23 @@ Pathfinder::Pathfinder (
         mDstAmount (saDstAmount),
         mSrcCurrency (uSrcCurrency),
         mSrcIssuer (uSrcIssuer),
-        mSrcAmount ({uSrcCurrency, uSrcIssuer}, 1u, 0, true),
+        convert_all_ (mDstAmount ==
+            STAmount (mDstAmount.issue (), 1u, 0, true)),
         mLedger (cache->getLedger ()),
         mRLCache (cache)
 {
-    assert (isXRP(uSrcCurrency) == isXRP(uSrcIssuer));
-}
+    if (! srcAmount)
+    {
+        mSrcAmount = STAmount({ uSrcCurrency,
```

Comment:
If no source amount is specified, create a default source amount. The default source amount specifies to spend as much as needed/possible. If the source currency is not XRP, the source amount issuer is the sender.